Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -521,6 +521,22 @@ Bytecode::CodeGenerationErrorOr<void> Identifier::generate_bytecode(Bytecode::Ge
     return {};
 }
 
+Bytecode::CodeGenerationErrorOr<void> SuperCall::generate_bytecode(Bytecode::Generator& generator) const
+{
+    Vector<Bytecode::Register> argument_registers;
```

Comment:
This is not correct for the (new) synthetic super call variant.
Which we use when you don't supply a constructor but have a parent class.
It should behave mostly like
```js
constructor(...args) {
    super(...args);
}
```

However we can't use that as you can poison `...args` by overwriting something in Array.

I don't think we need this right now so you could just check for it and return an error something like:
```cpp
return Bytecode::CodeGenerationError {
    this,
    "Synthetic super call not supported yet"sv
};
```

Reply:
I think I can manage to tinker something that gets this done with what is already written in the AST version